### PR TITLE
1679 dataset sorce and filter source@devel

### PR DESCRIPTION
**Pull request by PoojaKumari05**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by bms63** [comment]
I think you will need to update the related unit tests.

**Review by bms63 — COMMENTED** [review]
Hi Pooja.  Can you confirm that none of our vignettes and templates need this update for these functions?

File: R/derive_param_extreme_event.R [hunk 1]
```diff
@@ -89,8 +89,8 @@
 #'
 derive_param_first_event <- function(dataset,
                                      dataset_adsl,
-                                     dataset_source,
-                                     filter_source,
+                                     dataset_add,
+                                     filter_add,
```
Comment (bms63) [inline R/derive_param_extreme_event.R:93]: We are going to need a deprecation warning message telling users to update accordingly - https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#parameter

Comment (bms63) [reply R/derive_param_extreme_event.R:93]: I'm not recalling functions where we have deprecated specific arguments off the top of my head - @pharmaverse/admiral does any remember a function where we did this sothat @PoojaKumari05 can search through GitHub change history for an example??

Comment (bms63) [reply R/derive_param_extreme_event.R:93]: Apologies these comments apply to  `derive_param_extreme_event` as `derive_param_first_event` is deprecated.  You actually don't need to change anything for `derive_param_first_event` as no one can use it.

Comment (bundfussr) [reply R/derive_param_extreme_event.R:93]: We renamed some arguments in admiral 0.3.0. But this was a long time ago. So maybe just follow the programming strategy?

**Comment by bms63** [comment]
Hey Pooja,

Let's pick this back up in the next release.  Going to close for now.

**Pull request closed without merging**